extensions/amp-link-rewriter/0.1/link-rewriter.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getConfigOpts} from './config-options';
+import {getDataParamsFromAttributes} from '../../../src/dom';
+
+export const
+    CTX_ATTR_NAME = 'shiftedctx',
+    CTX_ATTR_VALUE = () => {
+      return Date.now();
```

Comment:
could you help me understand what this timestamp is trying to do?

Reply:
We use this value (date) as a flag to mark element as rewritten so we prevent double rewriting

Reply:
so does the value really matter? 
also, we would namespace the attribute name: `i-amphtml-link-rewritten`

Reply:
this functionality was meant to be used for debug purposes on secondary click. On the context click you would get link rewritten and it will revert back to original after a short time (10 secs). Purpose was link rewriting preview.
Now thinking, this might not be useful at all in amp context. So we are thinking about removing it completely.

Reply:
rewriting on "contextmenu" event is now removed